src/v/cloud_storage/cache_service.cc

```diff
@@ -1383,9 +1393,26 @@ cache::trim_carryover(uint64_t delete_bytes, uint64_t delete_objects) {
     co_return result;
 }
 
+void cache::maybe_background_trim() {
+    bool bytes_over_limit = _current_cache_size + _reserved_cache_size
+                            > _trim_threshold_bytes();
+    bool objects_over_limit = _current_cache_objects + _reserved_cache_objects
+                              > _trim_threshold_objects();
+
+    if (bytes_over_limit || objects_over_limit) {
+        vlog(cst_log.debug, "Spawning background trim_throttled");
+        ssx::spawn_with_gate(_gate, [this]() -> ss::future<> {
```

Comment:
please remove the coroutine lambda, the lambda could just return the result of `trim_throttled`

Reply:
co-routine lambdas are unfortunately a big footgun. CI correctly caught this in static analysis https://buildkite.com/redpanda/redpanda/builds/49800#018fdfc4-5da2-4dd9-89d7-717e09ed56f5/11990-13081